### PR TITLE
Themes: Remove the `bind`s in `getButtonOptions`

### PR DESCRIPTION
**Pull request by seear**
Using Chrome's Timeline View whilst scrolling through the themes at http://calypso.localhost:3000/design, we can see that the frame rate regularly drops to around 1 frame-per-second. The frame rate needs to stay at around 60fps for a smooth scroll.

screen shot 2016-01-12 at 15 39 20

* Taking measurements on the Master branch in dev mode, the longest frames are around 1373ms
* By removing the call to `getButtonOptions` in [`themes-list`](https://github.com/Automattic/wp-calypso/blob/928fd8c74058bafbf929c8fbed2b521409a813ca/client/components/themes-list/index.jsx#L56), the longest frames can be reduced to around 383ms, which whilst not perfect makes a large difference to perceived performance

This PR is an attempt to optimise whatever is happening in `getButtonOptions` that causes the janky scroll.

**To Do**
- [ ] Cache translations in `getButtonOptions`

### CODE REVIEW

File: shared/components/theme/more-button.jsx [hunk 1]
```diff
@@ -92,7 +92,7 @@ var ThemeMoreButton = React.createClass( {
 							);
 						}
 						return (
-							<PopoverMenuItem key={ option.label } action={ option.action }>
+							<PopoverMenuItem key={ option.label } action={ this.state.showPopover ? option.action() : null }>
```
Comment (mcsf) [inline shared/components/theme/more-button.jsx]: Drive-by, contextless comment: is the invocation on `action` (`()`) intentional or accidental?

Comment (mcsf) [inline shared/components/theme/more-button.jsx]: Disregard, I just saw the rest. :) (I'll delete this)

**Comment by seear** [comment]
With current changes, performance is no better than master, perhaps even worse, with frames 1800ms in length. I suspect that the closure that has been introduced performs no better that the binds.

**Comment by seear** [comment]
`getTranslationFromJed` shows up as expensive in the timeline (`ENV=wpcalypso`), so caching the translations is a simple optimisation we can make.

**Comment by seear** [comment]
Closed in favour of #2395 and other refactors such as #2444
